python/cuml/test/test_dbscan.py

```diff
@@ -48,10 +51,13 @@ def test_dbscan_predict(datatype, input_type):
 
     for i in range(X.shape[0]):
         assert cu_labels[i] == sk_labels[i]
+    if stream is not None:
+        stream.sync()
```

Comment:
In case stream is None, `handle` will use the default stream internally right? We still need to synchronize in that case as the C++ API does not guarantee that the work is done after the call returns.

Reply:
Fair point. I have added a `sync` method to the `Base` itself to safely issue the synchronization call. Also updated these tests. However, I'd like @jirikraus @dantegd @cjnolet to comment on this method. Is this OK?

Reply:
I am not sure if this is OK as I don't think I understand the motivation for `base.sync()` correctly. Please see my other review comments.

Reply:
This is the same conversation as being done in [here](https://github.com/rapidsai/cuml/pull/475#discussion_r275665989). Resolving this one on that behalf.